client/interface_veth_test.go

```diff
@@ -0,0 +1,62 @@
+package client
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestFindInterfaceVeth_onNonExistantInterfaceVeth(t *testing.T) {
+	SkipIfRouterOSV6OrEarlier(t, sysResources)
+	c := NewClient(GetConfigFromEnv())
+
+	name := "Interface Veth does not exist"
+	_, err := c.FindInterfaceVeth(name)
+
+	require.Truef(t, IsNotFoundError(err),
+		"Expecting to receive NotFound error for Interface Veth %q.", name)
+}
+
+func TestAddFindDeleteInterfaceVeth(t *testing.T) {
+	SkipIfRouterOSV6OrEarlier(t, sysResources)
+	c := NewClient(GetConfigFromEnv())
+
+	name := "new_interface_veth"
+	interfaceVeth := &InterfaceVeth{
+		Name:       name,
+		Disabled:   false,
+		Address: 	"192.168.88.2/24",
+		Gateway:    "192.168.88.1",
+		Comment:    "new interface from test",
+	}
+
+	created, err := c.Add(interfaceVeth)
+	if err != nil {
```

Comment:
I copied and modified these files from the wireguard and dns pool client and resource items.  Is there an example within the project that lines up with what you are suggesting I change to.  I have also looked at the dns files like suggested below for the codegen, they match up in much the same way as the files I submitted.

Reply:
@robbisso not all tests are updated, but it would be great to have new code aligned.
The [vlan_interface_test.go](https://github.com/ddelnano/terraform-provider-mikrotik/blob/master/client/vlan_interface_test.go) contains only `assert/require` statements.
If you need help with code, I can create `suggestions` in this PR.